### ISSUE TITLE
Master branch compilation error with Intel (17.1/18.2) compiler

### ISSUE THREAD

**Issue by weixie4** [aoterodelaroza/critic2#16]
Dear Dr. Otero-de-la-Roza,

I wanted to bring it to your attention that when I tried to compile the master branch with Intel compilers (both version 17.1 and 18.2) I got the following error:

doqhull.c(39): error: expected an expression
    for (int i = 0; i < n; i++)
         ^

doqhull.c(39): error: identifier "i" is undefined
    for (int i = 0; i < n; i++)

The stable branch worked fine.  

Thanks,
Wei

**Comment by aoterodelaroza** [comment]
Hi Wei,

I just tried with 17.1 and, besides a duplicate variable in
flux@proc.f90 (please git pull the new version), it worked. Can you
please send the configure line you are using? Also, the output of
configure should say something like this:

configure: FC        = ifort
configure: F77       = ifort
configure: CC        = icc
configure: CXX       = g++
configure: FCFLAGS   = -g -FR -fopenmp
configure: FFLAGS    = -g 
configure: CFLAGS    = -g -O2 -Iqhull/
configure: CXXFLAGS  = -g -O2
configure: LDFLAGS   = 
configure: LIBS      = 
configure: LIBADD    =  qhull/libqhull_critic.a

What does it say for you in the CC line?

Alberto

**Comment by aoterodelaroza** [comment]
Sorry for the wait. I think the problem is declaring a variable inside
the loop is from the C99 standard, and some C compilers may need an
additional flag to allow it (though strangely my icc seems to go with
it). Normally I would change the code to conform to the older standard
(it's just syntactic sugar after all), but unfortunately the qhull
library also has this problem and I'd rather not touch it.

In summary, I have made a change to the configure.ac and pushed it to
the development version. Could you please do:

autoreconf -i
./configure
make clean
make

to try again? Failing that, you can try to add the corresponding
compiler flag to CFLAGS, which with icc I think is "-std=c99":

CC=icc CFLAGS='-std=c99' ./configure

**Comment by weixie4** [comment]
Now it worked like a charm. Thanks a lot for your help, Alberto. 
A side note: the CFLAGS given by configure is '-std=gnu99'.

**Comment by aoterodelaroza** [comment]
No worries. :)
